### PR TITLE
Configure linters to comply with Drake C++ style

### LINKED ISSUE #15: Add drake style linter configuration to drake-ros

### PR DESCRIPTION
**Pull request by hidmic**
Closes #15. Precisely what the title says. This patch configures `ament_cpplint` and `ament_clang_format` linters to match [drake](https://github.com/RobotLocomotion/drake)'s style (pulling in their configuration).

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/16)

### CODE REVIEW

**Comment by hidmic** [comment]
@IanTheEngineer there's also a `clang-tidy` configuration in [RobotLocomotion/drake](https://github.com/RobotLocomotion/drake), which would need extra steps to be configured here but _should_ be possible.

**Review by sloretz — COMMENTED** [review]
One nitpick and these changes look good to me!

I was going to suggest using Drake's `.clang-tidy` config too, but after trying that myself locally it's clear that Drake doesn't actually use it. https://github.com/RobotLocomotion/drake/pull/15654

File: drake_ros_core/.clang-format [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    Priority: 20
+  # C++ system headers (as of C++17).
+  - Regex:    '^[<"](algorithm|any|array|atomic|bitset|cassert|ccomplex|cctype|cerrno|cfenv|cfloat|charconv|chrono|cinttypes|ciso646|climits|clocale|cmath|codecvt|complex|condition_variable|csetjmp|csignal|cstdalign|cstdarg|cstdbool|cstddef|cstdint|cstdio|cstdlib|cstring|ctgmath|ctime|cuchar|cwchar|cwctype|deque|exception|execution|filesystem|forward_list|fstream|functional|future|initializer_list|iomanip|ios|iosfwd|iostream|istream|iterator|limits|list|locale|map|memory|memory_resource|mutex|new|numeric|optional|ostream|queue|random|ratio|regex|scoped_allocator|set|shared_mutex|sstream|stack|stdexcept|streambuf|string|string_view|strstream|system_error|thread|tuple|type_traits|typeindex|typeinfo|unordered_map|unordered_set|utility|valarray|variant|vector)[">]$'
+    Priority: 30
+  # Other libraries' h files (with angles).
+  - Regex:    '^<'
+    Priority: 40
+  # Your project's h files.
+  - Regex:    '^"drake'
```
[the remark this answers is not in the document]
Comment (hidmic) [reply drake_ros_core/.clang-format]: Good catch! See https://github.com/RobotLocomotion/drake-ros/pull/16/commits/e08194561b3c93df1a3f804261ec830a3776aa58.

**Comment by hidmic** [comment]
Rebased to solve conflicts after #17.

**Pull request merged**